packages/steiger-plugin-fsd/src/forbidden-imports/index.spec.ts

```diff
@@ -16,103 +28,158 @@ vi.mock('node:fs', async (importOriginal) => {
 
   return createFsMocks(
     {
-      '/shared/ui/styles.ts': '',
-      '/shared/ui/Button.tsx': 'import styles from "./styles";',
-      '/shared/ui/TextField.tsx': 'import styles from "./styles";',
-      '/shared/ui/index.ts': '',
-      '/entities/user/ui/UserAvatar.tsx': 'import { Button } from "@/shared/ui"',
-      '/entities/user/index.ts': '',
-      '/entities/product/ui/ProductCard.tsx': 'import { UserAvatar } from "@/entities/user"',
-      '/entities/product/index.ts': '',
-      '/features/comments/ui/CommentCard.tsx': 'import { styles } from "@/pages/editor"',
-      '/features/comments/index.ts': '',
-      '/pages/editor/ui/styles.ts': '',
-      '/pages/editor/ui/EditorPage.tsx': 'import { Button } from "@/shared/ui"; import { Editor } from "./Editor"',
-      '/pages/editor/ui/Editor.tsx': 'import { TextField } from "@/shared/ui"',
-      '/pages/editor/index.ts': '',
+      '/src/shared/ui/styles.ts': '',
+      '/src/shared/ui/Button.tsx': 'import styles from "./styles";',
+      '/src/shared/ui/TextField.tsx': 'import styles from "./styles";',
+      '/src/shared/ui/index.ts': '',
+      '/src/entities/user/ui/UserAvatar.tsx': 'import { Button } from "@/shared/ui"',
+      '/src/entities/user/index.ts': '',
+      '/src/entities/product/ui/ProductCard.tsx': 'import { UserAvatar } from "@/entities/user"',
+      '/src/entities/product/index.ts': '',
+      '/src/entities/cart/ui/SmallCart.tsx': 'import { App } from "@/app"',
+      '/src/entities/cart/lib/count-cart-items.ts': 'import root from "@/app/root.ts"',
+      '/src/entities/cart/lib/index.ts': '',
+      '/src/entities/cart/index.ts': '',
+      '/src/features/comments/ui/CommentCard.tsx': 'import { styles } from "@/pages/editor"',
+      '/src/features/comments/index.ts': '',
+      '/src/pages/editor/ui/styles.ts': '',
+      '/src/pages/editor/ui/EditorPage.tsx': 'import { Button } from "@/shared/ui"; import { Editor } from "./Editor"',
+      '/src/pages/editor/ui/Editor.tsx': 'import { TextField } from "@/shared/ui"',
+      '/src/pages/editor/index.ts': '',
+      '/src/app': '',
+      '/src/app/ui/index.ts': '',
+      '/src/app/index.ts': '',
+      '/src/app/root.ts': '',
     },
     originalFs,
   )
 })
 
 it('reports no errors on a project with only correct imports', async () => {
   const root = parseIntoFsdRoot(`
-    📂 shared
-      📂 ui
-        📄 styles.ts
-        📄 Button.tsx
-        📄 TextField.tsx
-        📄 index.ts
-    📂 pages
-      📂 editor
+    📂 src
```

Comment:
issue: the idea is to give the `check` function of rules the folder of the FSD root, so by putting the layers in `src`, we're not actually testing how it's going to be

suggestion: we can add a second optional argument to `parseIntoFsdRoot` — the root path of the folders, then we can avoid having `src` here in the structure

Reply:
ah, now I see that in other tests, you access `.children[0]` to overcome this issue. That's also fine, although you forgot to do it in the positive case. I still think that it would be cleaner with an optional root path argument though

Reply:
Yeah, I had to put all the files into /src because of the problem I wrote about in the PR description. Indeed I forgot to add `.children[0]` for the positive case, thanks for catching that. 

I would be happy to pass that optional argument to `parseIntoFsdRoot` but I see that it accepts only 1 argument. Maybe you implemented the second parameter in some branch that has not yet been merged to the master?

<img width="793" alt="Screenshot 2024-07-23 at 22 36 08" src="https://github.com/user-attachments/assets/304f7987-4c37-46e6-aca2-f3c3ad58812e">

Reply:
I meant to suggest for you to add that second parameter :)

Reply:
I also took the liberty to unresolve this conversation. Usually, I prefer to mark threads as resolved when the comment was either fully addressed or it was decided not to address it, but this comment is still relevant partially. Let me know if you'd like to add that second parameter, or I can do it

Reply:
Oh, got it. No, problem, I'll add that tomorrow. I think it will be useful in further test cases and also I don't like `.children[0]` too much.